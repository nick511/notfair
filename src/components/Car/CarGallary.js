import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import Slider from 'react-slick'
import ReactImageFallback from 'react-image-fallback'

import 'slick-carousel/slick/slick.css'
import 'slick-carousel/slick/slick-theme.css'
import 'style/slick-overwrite.scss'
import styles from './CarGallary.module.scss'

class CarGallary extends PureComponent {
  state = {
    activeSlide: 1,
  }

  _beforeChange = (current, next) => {
    this.setState({ activeSlide: next + 1 })
  }

  render() {
    return (
      <div>
        <Slider beforeChange={this._beforeChange}>
          {this.props.images.map(image => (
            <div key={image}>
              <ReactImageFallback
                src={image}
                fallbackImage="/images/car-image-placeholder.png"
                alt=""
                className={styles.mainImage}
              />
            </div>
          ))}
        </Slider>

        {/* Image page info */}
        <div className={styles.pageInfo}>
          {this.state.activeSlide} / {this.props.images.length}
        </div>
      </div>
    )
  }
}

CarGallary.propTypes = {
  images: PropTypes.array.isRequired,
}

export default CarGallary
