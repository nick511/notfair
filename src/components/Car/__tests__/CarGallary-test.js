import React from 'react'
import { shallow } from 'enzyme'

import CarGallary from '../CarGallary'

describe('<CarGallary />', () => {
  let wrapper = null
  const images = [
    'http://cdn-prod.prod.fair.engineering/vehicle-images/0640_032_png/MY2016/10846/10846_cc0640_032_BS.png',
    'https://content.homenetiol.com/2001593/2112019/640x480/28ed530709334ec58855b2a2fbf3df3b.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/21010f808ea144fea0d73441c2593a6d.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/4eb8ef65ed5d4b4a9dc3b7c81cf78b24.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/d8565261a3544f07a56897c602152a60.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/ebf7b28a6d5545bdab204200f1c655c3.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/ca2df8b0532543269a3807e08f3cd2f4.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/ae54d1a9caf84f5b9a99819335b124f9.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/bd8c3a219a1a40ae9d0f34416b72cd06.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/a42226cc586f4e6486a34337f3b86cc1.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/b45d421d054f4c4da90669a20f029bd5.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/48458b96bb0b4ba28d8f9be7c0e9af71.jpg',
    'https://content.homenetiol.com/2001593/2112019/640x480/2c3bd441b6f54d5f9358e3fb4e164397.jpg',
  ]

  beforeEach(() => {
    wrapper = shallow(<CarGallary images={images} />)
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders all images', () => {
    const nodes = wrapper.find('.mainImage')
    expect(nodes).toHaveLength(images.length)
    nodes.forEach((node, index) => {
      expect(node.prop('src')).toEqual(images[index])
    })
  })

  it('chages page number after sliding', () => {
    wrapper.instance()._beforeChange(0, 1)
    expect(wrapper.state('activeSlide')).toBe(2)
    expect(wrapper).toMatchSnapshot()
  })
})
