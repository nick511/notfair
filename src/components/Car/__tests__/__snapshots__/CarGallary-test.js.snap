// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarGallary /> chages page number after sliding 1`] = `
<div
  className="component"
>
  <Slider
    beforeChange={[Function]}
    className="slider"
  >
    <div
      key="http://cdn-prod.prod.fair.engineering/vehicle-images/0640_032_png/MY2016/10846/10846_cc0640_032_BS.png"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="http://cdn-prod.prod.fair.engineering/vehicle-images/0640_032_png/MY2016/10846/10846_cc0640_032_BS.png"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/28ed530709334ec58855b2a2fbf3df3b.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/28ed530709334ec58855b2a2fbf3df3b.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/21010f808ea144fea0d73441c2593a6d.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/21010f808ea144fea0d73441c2593a6d.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/4eb8ef65ed5d4b4a9dc3b7c81cf78b24.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/4eb8ef65ed5d4b4a9dc3b7c81cf78b24.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/d8565261a3544f07a56897c602152a60.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/d8565261a3544f07a56897c602152a60.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/ebf7b28a6d5545bdab204200f1c655c3.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/ebf7b28a6d5545bdab204200f1c655c3.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/ca2df8b0532543269a3807e08f3cd2f4.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/ca2df8b0532543269a3807e08f3cd2f4.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/ae54d1a9caf84f5b9a99819335b124f9.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/ae54d1a9caf84f5b9a99819335b124f9.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/bd8c3a219a1a40ae9d0f34416b72cd06.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/bd8c3a219a1a40ae9d0f34416b72cd06.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/a42226cc586f4e6486a34337f3b86cc1.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/a42226cc586f4e6486a34337f3b86cc1.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/b45d421d054f4c4da90669a20f029bd5.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/b45d421d054f4c4da90669a20f029bd5.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/48458b96bb0b4ba28d8f9be7c0e9af71.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/48458b96bb0b4ba28d8f9be7c0e9af71.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/2c3bd441b6f54d5f9358e3fb4e164397.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/2c3bd441b6f54d5f9358e3fb4e164397.jpg"
      />
    </div>
  </Slider>
  <div
    className="pageInfo"
  >
    2
     / 
    13
  </div>
</div>
`;

exports[`<CarGallary /> renders without crashing 1`] = `
<div
  className="component"
>
  <Slider
    beforeChange={[Function]}
    className="slider"
  >
    <div
      key="http://cdn-prod.prod.fair.engineering/vehicle-images/0640_032_png/MY2016/10846/10846_cc0640_032_BS.png"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="http://cdn-prod.prod.fair.engineering/vehicle-images/0640_032_png/MY2016/10846/10846_cc0640_032_BS.png"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/28ed530709334ec58855b2a2fbf3df3b.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/28ed530709334ec58855b2a2fbf3df3b.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/21010f808ea144fea0d73441c2593a6d.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/21010f808ea144fea0d73441c2593a6d.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/4eb8ef65ed5d4b4a9dc3b7c81cf78b24.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/4eb8ef65ed5d4b4a9dc3b7c81cf78b24.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/d8565261a3544f07a56897c602152a60.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/d8565261a3544f07a56897c602152a60.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/ebf7b28a6d5545bdab204200f1c655c3.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/ebf7b28a6d5545bdab204200f1c655c3.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/ca2df8b0532543269a3807e08f3cd2f4.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/ca2df8b0532543269a3807e08f3cd2f4.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/ae54d1a9caf84f5b9a99819335b124f9.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/ae54d1a9caf84f5b9a99819335b124f9.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/bd8c3a219a1a40ae9d0f34416b72cd06.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/bd8c3a219a1a40ae9d0f34416b72cd06.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/a42226cc586f4e6486a34337f3b86cc1.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/a42226cc586f4e6486a34337f3b86cc1.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/b45d421d054f4c4da90669a20f029bd5.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/b45d421d054f4c4da90669a20f029bd5.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/48458b96bb0b4ba28d8f9be7c0e9af71.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/48458b96bb0b4ba28d8f9be7c0e9af71.jpg"
      />
    </div>
    <div
      key="https://content.homenetiol.com/2001593/2112019/640x480/2c3bd441b6f54d5f9358e3fb4e164397.jpg"
    >
      <ReactImageFallback
        alt=""
        className="mainImage"
        fallbackImage="/images/car-image-placeholder.png"
        initialImage={null}
        src="https://content.homenetiol.com/2001593/2112019/640x480/2c3bd441b6f54d5f9358e3fb4e164397.jpg"
      />
    </div>
  </Slider>
  <div
    className="pageInfo"
  >
    1
     / 
    13
  </div>
</div>
`;
